cachi2/interface/cli.py

```diff
@@ -102,6 +104,25 @@ class _Input(pydantic.BaseModel, extra="forbid"):
     flags: list[Flag] = list()
 
 
+def _yaml_validate_set_config(file: str) -> str:
+
+    try:
+        with open(file, "r") as value:
+            config.config = config.Config(**yaml.safe_load(value))
+    except yaml.YAMLError:
+        raise typer.BadParameter(f"Not valid YAML: {file!r}")
+    except FileNotFoundError:
+        config.config = config.Config()
+        return ""
+    except TypeError as e:
+        if str(e) == "expected str, bytes or os.PathLike object, not NoneType":
+            config.config = config.Config()
+            return ""
```

Comment:
You get this type error by trying to `open(None)`. Instead, if the user doesn't use `--config-file` (which means the file path is None), we shouldn't try to open anything.